src/htm/algorithms/Connections.hpp

```diff
@@ -374,7 +378,8 @@ class Connections : public Serializable
    *
    * @retval Synapse data.
    */
-  const SynapseData &dataForSynapse(const Synapse synapse) const {
+  inline const SynapseData& dataForSynapse(const Synapse synapse) const {
+    NTA_CHECK(synapseExists_(synapse, true));
```

Comment:
needed check for `dataForSynapse/Segment()`. Otherwise, the data could be accessed even after removing the synapse/segment with `destroy*()`.